uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -131,14 +131,19 @@ private static function _extractPreviousAssignee(&$assignees, $previousAssigneeI
    * Gets the full list of the recipients of the reminder email
    * Makes sure there are no duplicates or no empty emails in the list
    *
-   * @param array $activityContacts
+   * @param array $contacts
    * @param array $previousAssignee
    * @return array
    */
-  private static function _reminderRecipients($activityContacts, $previousAssignee) {
-    return array_filter(array_unique(array_merge(
-                            $activityContacts['assignees']['emails'], $activityContacts['source']['emails'], [$previousAssignee['email']]
-    )));
+  private static function _reminderRecipients($contacts, $previousAssignee) {
+    $hasSources = !empty($contacts['source']['emails']);
```

Comment:
a shorter way to write this would be :

```php
$assignees = $contacts['assignees']['emails'] ?: [];
$sources = $contacts['source']['emails'] ?: [];
$previous = !empty($previousAssignee['email']) ? [$previousAssignee['email']] : [];

$recipients = array_merge($assignees, $sources, $previous);

return array_filter(array_unique($recipients));
```

hence in your code that you forget to the 'email' index for $previous  variable , also you miss some spaces that make the code easier to read ... both corrected in my version